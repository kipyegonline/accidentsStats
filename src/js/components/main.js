import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import styled, { createGlobalStyle } from "styled-components";
import { Row, Col } from "reactstrap";
import PropTypes from "prop-types";
import * as d3 from "d3";
import TableStats from "./Table/Table";
import Layout from "./UI/Layout";
import { months } from "./tools/SvgComp";

function Main() {
  const [selected, setSelected] = useState(0);

  const data = useSelector((state) => {
    return {
      ntsa: state.ntsa,
      ntsb: state.ntsb,
      loadedA: state.loadedA,
      loadedB: state.loadedB,
    };
  });

  const fixData = (data) => {
    let main = [];
    for (let i = 0; i < data.length; i += 6) {
      main = [...main, data.slice(i, i + 6)];
    }

    return main;
  };

  const main = fixData(data.ntsa);
  const mainB = fixData(data.ntsb);
  const getValue = (val) => setSelected(val);
  let month = "";

  return (
    <Layout>
      <Row className="my-3">
        <Row className="mt-5">
          <Col>
            {data.loadedA && data.loadedB ? (
              <SelectedMonth data={main} sendValue={getValue} />
            ) : null}
          </Col>
        </Row>
        <Col className="position-relative">
          <p>
            COMPARATIVE STATISTICS TRENDS FOR 2019 AND 2020 AS AT {"  "}
            <span className="text-info">
              {" "}
              {month === "Invalid Date" ? "" : month}{" "}
            </span>
          </p>
          {data.loadedA && data.loadedB ? (
            <TableStats ntsa={main[selected]} />
          ) : (
            <p>Loading</p>
          )}
        </Col>
        <Col className="position-relative">
          <p>
            COMPARATIVE STATISTICS TRENDS FOR 2019 AND 2020 {"  "}
            <span className="text-info">
              {" "}
              {month === "Invalid Date" ? "" : month}{" "}
            </span>
          </p>
          {data.loadedA && data.loadedB ? (
            <TableStats ntsa={mainB[selected]} />
          ) : (
            <p>Loading</p>
          )}
        </Col>
      </Row>
    </Layout>
  );
}
export default Main;

export const SelectedMonth = ({ data, sendValue }) => {
  const [month, setMonth] = useState(data.length - 1);

  const getMonth = (d) => new Date(d).getMonth();
  const handleChange = (d) => {
    if (d >= 0) {
      sendValue(d);

      setMonth(d);
    }
  };
  useEffect(() => {
    sendValue(month);
  }, []);

  return (
    <select
      className="form-control"
      onChange={(e) => handleChange(Number(e.target.value))}
      value={month}
    >
      {data.map((d, i) => (
        <option key={i} value={i}>
          {months[getMonth(d[0].addedon)]}
        </option>
      ))}
    </select>
  );
};
SelectedMonth.propTypes = {
  data: PropTypes.arrayOf(PropTypes.array),
  sendValue: PropTypes.func,
};
