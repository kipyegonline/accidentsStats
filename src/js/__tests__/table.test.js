import React, { Component } from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TableStats from "../components/Table/Table";
import { SelectedMonth } from "../components/main";
import {
  TableBody,
  TableList,
  TableFoot,
  TableHeader,
} from "../components/Table/TableList";

Enzyme.configure({ adapter: new Adapter() });
export const data = [
  {
    addedon: "2020-02-14",
    id: 1,
    fatalties: 345,
    slightlyInjured: 435,
    seriouslyInjured: 654,
    victimClass: "drivers",
    fill: "green",
  },

  {
    addedon: "2020-03-14",
    id: 2,
    fatalties: 245,
    slightlyInjured: 405,
    seriouslyInjured: 554,
    victimClass: "pillion_passengers",
    fill: "yellow",
  },
  {
    addedon: "2020-05-14",
    id: 3,
    fatalties: 445,
    slightlyInjured: 535,
    seriouslyInjured: 454,
    victimClass: "pedestrians",
    fill: "red",
  },
];
/*
describe("<Select /> Component", () => {
  let _change = jest.fn();
  const wrapper = shallow(<SelectedMonth sendValue={_change} data={data} />);

  test("presence of component", () =>
    expect(wrapper.find(".form-control")).toHaveLength(1));
  it("has text", () =>
    expect(
      wrapper
        .find("option")
        .at(1)
        .text()
    ).toBe("February"));
  it("changes on changes", () => {
    wrapper.find(".form-control").simulate("change", { target: { value: 0 } });

    expect(_change).toBeCalled();
    expect(_change).toBeCalledWith(0);
    expect(_change).toBeCalledTimes(1);
  });
  it("shows current undefined value when not selected", () =>
    expect(wrapper.find(".form-control").prop("value")).toBeUndefined());
});*/
describe("<TableList/>", () => {
  const wrapper = mount(
    <table>
      <tbody>
        <tr>
          <TableList data={data} a="fatalties" TableBody={TableBody} />
        </tr>
      </tbody>
    </table>
  );
  it("renders", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  //console.log(wrapper.debug());
});
