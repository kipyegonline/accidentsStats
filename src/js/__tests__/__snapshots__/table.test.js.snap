// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableList/> renders 1`] = `"<table><tbody><tr><td class=\\"drivers-fatalties-2020\\">345</td><td class=\\"pillion_passengers-fatalties-2020\\">245</td><td class=\\"pedestrians-fatalties-2020\\">445</td></tr></tbody></table>"`;
